app/main/browser-window.js

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2016 Mozilla
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+this file except in compliance with the License. You may obtain a copy of the
+License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+*/
+
+import path from 'path';
+import electron from 'electron';
+
+import * as hotkeys from './hotkeys';
+import { UI_DIR, fileUrl } from './util';
+import BUILD_CONFIG from '../../build-config';
+
+const BROWSER_CHROME_URL = fileUrl(path.join(UI_DIR, 'browser', 'browser.html'));
+const BrowserWindow = electron.BrowserWindow;  // create native browser window.
+
+/**
+ * Top-level list of Browser Windows.
+ * @TODO Is this still needed? Electron can return all of its browser windows.
+ */
+
+const browserWindows = [];
+
+/**
+ * Take a promise to `ProfileStorage`. Remove this once
+ * UA service is in its own process. Also takes an onload callback --
+ * currently used for the first window created to record load times.
+ */
+export async function createBrowserWindow(profileStoragePromise, onload) {
```

Comment:
It appears we're always invoking `createBW` in an async function.  If so, just pass `profileStorage` -- or consider passing the window scope directly, moving the profile usage out to the caller, which is probably closer to what we want over time.